src/cpp/src/group_beam_searcher.cpp

```diff
@@ -385,7 +369,21 @@ EncodedResults beam_search(ov::InferRequest& lm,
         prompts.push_back(std::vector<int64_t>{prompt_start, prompt_start + sequence_length});
     }
 
-    initialize_inputs(input_ids, attention_mask, lm);
+    lm.set_tensor("input_ids", input_ids);
+    lm.set_tensor("attention_mask", attention_mask);
+    bool position_ids_available = position_ids.has_value();
+    if (position_ids_available)
+        lm.set_tensor("position_ids", *position_ids);
+
+    ov::Tensor beam_idx = lm.get_tensor("beam_idx");
+    beam_idx.set_shape({batch_size});
+    auto beam_data = beam_idx.data<int32_t>();
+    if (selected_beam_idx.has_value()) {
+        OPENVINO_ASSERT((*selected_beam_idx).size() == 1);
+        beam_data[0] = (*selected_beam_idx).at(0);
```

Comment:
`OPENVINO_ASSERT(batch_size == 1, "continuation of generation is possible only for batch 1");` feels completely arbitrary when it's in `generate()`. Moving it there would explain that it's not possible because you can't fill the rest of `beam_data`.

Reply:
I didn't quite understand. You mean it would be better to remove existing assert for batch_size == 1 from llm_pipeline and leave restriction here?

Anyways, i changed signature. Now only one selected beam idx (in case if num_return_sequences == 1) is returned and accepted, so this assert is removed.

Reply:
> I didn't quite understand. You mean it would be better to remove existing assert for batch_size == 1 from llm_pipeline and leave restriction here?

Yes

Reply:
Communicated offline. We have limitations that do not allow continuation for greedy as well
We drop batches which met eos 
https://github.com/openvinotoolkit/openvino.genai/blob/master/src/cpp/src/greedy_decoding.cpp#L128-L132
 Therefore assert is in the common `llm_pipeline.cpp`